### DISCUSSION TITLE
How do I access supabase from the backend node.js

### DISCUSSION THREAD

**Discussion by milestones95** [supabase/supabase#2125] [Questions] [answered]
I have an api that's called for stripe checkout, and only after the checkout is successfully completed I want to call insert a new order id with the user details into supabase. How can I do this from the backend so that a user can't manipulate the client call to fake a transaction?

Here's the api:
```js
app.post('/create-checkout-session', async (req, res) => {
       const {request} = req;
       const account_id = req.body.account_id;
       const user_id = req.body.user_id;
       var successLink = 'http://localhost:3000/dashboard/userreq/'+user_id;
       const session = await stripe.checkout.sessions.create({
       payment_method_types: ['card'],
         line_items: [
        {
         price_data: {
          currency: 'usd',
          product_data: {
            name: 'Story',
          },
          unit_amount: 1300,
        },
        quantity: 1,
      },
    ],
    payment_intent_data: {
    application_fee_amount: 123,
    transfer_data: {
      destination: account_id,
    },
  },
    mode: 'payment',
    success_url: successLink,
    cancel_url: 'http://localhost:3000/cancel.html',
  });
res.send({
  sessionId: session.id,
});});
```

**Comment by inian** [comment, marked as answer]
Hi @milestones95, you can use `supabase-js` on your Node server to [insert](https://supabase.io/docs/postgrest/client/insert) a new order into your database. Let me know if there are any issues with this approach.

**Reply by milestones95** [reply]
How does it know what supabase account and api key to use? I see where I plug in the URL of the site hosting..

**Reply by ashleyconnor** [reply]
If you want to insert as the user using RLS, then you need to pass the users JWT to your backend and configure the client to use it: `supabase.auth.setAuth(token)`.

**Reply by inian** [reply]
You can find the API url and key in the dashboard here - https://app.supabase.io/project/your-project-id/settings/api. You can use the service key on the server side.

**Reply by fergusmeiklejohn** [reply]
Are there any issues with using supabase-js in serverless functions?

**Reply by ashleyconnor** [reply]
>Are there any issues with using supabase-js in serverless functions?

The library claims to be isomorphic. No issues using it with Cloudflare Workers.

**Reply by bbugh** [reply]
The link https://supabase.io/docs/postgrest/client/insert is dead, could you link the updated one?

**Reply by inian** [reply]
https://supabase.com/docs/reference/javascript/insert this is the method for inserting data

**Comment by 24jr** [comment]
As of 2023, I believe `supabase.auth.setAuth(token)` as suggested above is not the way with v2. I'm looking at https://supabase.com/docs/reference/javascript/auth-getuser 
```
const { data: { session }, error } = await supabaseClient.auth.getSession()
 const response = await fetch('/api/user', {
          method: 'GET',
          headers: {
            'content-type': 'application/json',
            'Authorization': 'Bearer ' + session?.access_token
          }
   });
```
then in my sveltekit +server.js file where i have endpoint I'm trying this 
```
import { supabaseClient } from "$lib/supabaseClient"

export async function GET(event) {
  try {
    console.log('getUserData')
    const authHeader = event?.request?.headers?.get('Authorization')
    console.log('authHeader',authHeader)
    if(!authHeader){
      return new Response(JSON.stringify({ message: 'No Authorization Token Provided'}, { status: 401 }))
    }
    const res = await supabaseClient.auth.getUser(authHeader)
    console.log('userres',res)
```
This is my first time trying to make endpoint with sveltekit as well as call supabase from server, but I think I'm doing this right but getting error `AuthApiError: This endpoint requires a Bearer token`

I might actually need to set the client again in server like this but still get same thing. And I am logging the access token (that's the same as jwt correct?) so believe is putting "Bearer eyfjfjf...." in
```
import { createClient } from '@supabase/supabase-js'
import { env } from '$env/dynamic/public';

export async function GET(event) {
  try {
    console.log('getUserData')
    const authHeader = event?.request?.headers?.get('Authorization')
    console.log('authHeader',authHeader)
    if(!authHeader){
      return new Response(JSON.stringify({ message: 'No Authorization Token Provided'}, { status: 401 }))
    }
    const supabaseClient = createClient(env.PUBLIC_SUPABASE_URL, env.PUBLIC_SUPABASE_ANON_KEY, {
      auth: {
        autoRefreshToken: false,
        persistSession: false,
        detectSessionInUrl: false
      }
    })
    const res = await supabaseClient.auth.getUser(authHeader)
    console.log('userres',res)
```
Actually, I think I figured it out from https://supabase.com/docs/guides/auth/auth-helpers/sveltekit#server-side-data-fetching-with-rls
```
import { getSupabase } from '@supabase/auth-helpers-sveltekit'
import { json, redirect } from '@sveltejs/kit'

export async function GET(event) {
  const { session, supabaseClient } = await getSupabase(event)
  if (!session) {
    throw redirect(303, '/')
  }
  console.log('sessss', session)
  const user_id = session?.user?.id
  console.log('user_id',user_id)
  const { data, error } = await supabaseClient.from('person_settings')
  .select('*')
  .eq('id', user_id)
  .limit(1)
  .single()
  console.log('person_settings', data, error)

  return json({ data })
}
```
